pandas/core/dtypes/cast.py

```diff
@@ -630,7 +630,7 @@ def infer_dtype_from(val, pandas_dtype: bool = False):
     return infer_dtype_from_array(val, pandas_dtype=pandas_dtype)
 
 
-def infer_dtype_from_scalar(val, pandas_dtype: bool = False):
+def infer_dtype_from_scalar(val, pandas_dtype: bool = False) -> Tuple[DtypeObj, Any]:
```

Comment:
IIUC correctly, in master the return type could be an instance or a type. There is obviously merit in refining types, but what is the advantanges/disadvantages of always returning instances instead of always returning types.

Reply:
AFAICT it is almost never a conscious decision to return a type instead of a dtype, just the author wasnt thinking of the distinction